modules/subschema/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/subschema/package.scala

```diff
@@ -224,45 +224,75 @@ package object subschema {
     val additionalProperties: Schema => Schema =
       _.additionalProperties.collect({ case AdditionalPropertiesSchema(value) => value }).getOrElse(any)
 
-    val p1: List[(String, Schema)] = properties(s1)
-    val p2: List[(String, Schema)] = properties(s2)
+    def stripAnchorsFromSchemaList(list: List[(String, Schema)]) =
+      list.map{case (r, s) => (stripAnchors(r), s)}
 
-    val pp1: List[(String, Schema)] = patternProperties(s1)
-    val pp2: List[(String, Schema)] = patternProperties(s2)
+    val p1: List[(String, Schema)] = stripAnchorsFromSchemaList(properties(s1))
+    val p2: List[(String, Schema)] = stripAnchorsFromSchemaList(properties(s2))
```

Comment:
these shouldn't be regexes technically, but just plain property names, so I'm not sure we need stripping here. Although it will not cause any problems unless the property names contain anchors, in which case I have no idea what would happen even if we didn't strip anchors, so it's probably fine

Reply:
Hm, yeah, you are correct. I think we can leave them as is for now, and see how it goes